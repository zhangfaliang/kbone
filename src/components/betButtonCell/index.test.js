import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import BetButtonCell from './index';
import BetButton from '../betButton';
import Line from '../line';
const delay = (timeout, cb) => setTimeout(cb, timeout);
describe('BetButtonCell', () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(
      <BetButtonCell>
        <BetButton />
        <Line />
      </BetButtonCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('defaultClickBetBtn', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetButtonCell>
        <BetButton />
      </BetButtonCell>
    );
    const clickBetBtn = wrapper.prop('clickBetBtn');
    const click = jest.fn(clickBetBtn);
    wrapper
      .find('BetButtonCell')
      .find('BetButton')
      .simulate('click');
    expect(click()).toBe('clickBetBtn');
  });
  it('changeSp', () => {
    expect.assertions(1);

    const wrapper = mount(
      <BetButtonCell>
        <BetButton sp={delay(200, () => 1.6)} />
      </BetButtonCell>
    );
    const changeSp = wrapper.prop('changeSp');
    const click = jest.fn(changeSp);
    expect(click()).toBe('changeSp');
  });
  it('changeHandicap', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetButtonCell>
        <BetButton handicap={delay(200, () => 6.5)} />
      </BetButtonCell>
    );
    const changeHandicap = wrapper.prop('changeHandicap');
    const click = jest.fn(changeHandicap);
    expect(click()).toBe('changeHandicap');
  });
});
