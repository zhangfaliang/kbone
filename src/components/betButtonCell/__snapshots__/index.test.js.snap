// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetButtonCell base 1`] = `
<div
  class="push-inplay-betButtonCell"
>
  <div
    class="default-bet-button"
  >
    <span
      class="up"
    >
      <span
        class="iconfont icon-sheng"
      />
    </span>
    <span
      class="down"
    >
      <span
        class="iconfont icon-jiang"
      />
    </span>
    <div
      class="wrapSp"
    >
      <span
        class="sp"
      />
    </div>
  </div>
  <div
    class="row-line"
  />
</div>
`;
